client/session.go

```diff
@@ -1731,25 +1727,43 @@ func (s *session) selectBlocksForSeriesFromPeerBlocksMetadata(
 
 		// If all the peers have the same non-nil checksum, we pick the peer with the
 		// fewest attempts and fewest outstanding requests
-		if sameNonNilChecksum {
-			// Order by least attempts then by least outstanding blocks being fetched
-			blocksMetadataQueues = blocksMetadataQueues[:0]
-			for i := range currEligible {
-				insert := blocksMetadataQueue{
-					blocksMetadata: currEligible[i],
-					queue:          peerQueues.findQueue(currEligible[i].peer),
+		if singlePeer || sameNonNilChecksum {
+			// Prepare the reattempt peers metadata
```

Comment:
Perhaps add to this comment, i.e. something like "Prepare the reattempt peers metadata so any possible peer is a candidate to retry from"